go/tasks/plugins/presto/client/presto_client.go

```diff
@@ -0,0 +1,66 @@
+package client
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+
+	"github.com/lyft/flyteplugins/go/tasks/plugins/svc"
+
+	"time"
+
+	"github.com/lyft/flyteplugins/go/tasks/plugins/presto/config"
+)
+
+const (
+	httpRequestTimeoutSecs = 30
+	//AcceptHeaderKey          = "Accept"
+	//ContentTypeHeaderKey     = "Content-Type"
+	//ContentTypeJSON          = "application/json"
+	//ContentTypeTextPlain     = "text/plain"
+	//PrestoCatalogHeader      = "X-Presto-Catalog"
+	//PrestoRoutingGroupHeader = "X-Presto-Routing-Group"
+	//PrestoSchemaHeader       = "X-Presto-Schema"
+	//PrestoSourceHeader       = "X-Presto-Source"
+	//PrestoUserHeader         = "X-Presto-User"
+)
+
+type prestoClient struct {
+	client      *http.Client
+	environment *url.URL
+}
+
+type PrestoExecuteArgs struct {
+	RoutingGroup string `json:"routing_group,omitempty"`
+	Catalog      string `json:"catalog,omitempty"`
+	Schema       string `json:"schema,omitempty"`
+	Source       string `json:"source,omitempty"`
+}
+type PrestoExecuteResponse struct {
+	ID      string
+	Status  svc.CommandStatus
+	NextURI string
+}
+
+func (p *prestoClient) ExecuteCommand(
+	ctx context.Context,
+	queryStr string,
+	extraArgs interface{}) (interface{}, error) {
+
+	return PrestoExecuteResponse{}, nil
+}
+
+func (p *prestoClient) KillCommand(ctx context.Context, commandID string) error {
+	return nil
+}
+
+func (p *prestoClient) GetCommandStatus(ctx context.Context, commandID string) (svc.CommandStatus, error) {
+	return NewPrestoStatus(ctx, "UNKNOWN"), nil
+}
+
+func NewPrestoClient(cfg *config.Config) svc.ServiceClient {
```

Comment:
Oh, that's interesting. Normally from my experience in other langs (java, scala, rust, etc), you typically tie yourself to an interface instead of a specific implementation. In our case, I feel this will help when trying to swap out the Mozart client for one based on just Presto. Otherwise, won't we need to  make a copy  of the calling  code (ie. the executor) in the  private repo?

Reply:
Well, interfaces in Go are kind of the inverse in Java.
The idiomatic way is something like this:
`func MyFunc(m SpecificInterface) SpecificType`
So your function takes an interface of exactly the functions it needs and returns a specific type..

I do not understand what do you mean by copying calling code?

Reply:
So given that I made the client specific to Presto, my original comment is largely void now. 

What I was trying to say is that if we made the function return a specific type:
```
func NewNoopPrestoClient(cfg *config.Config) PrestoClient {
```

Then in the calling code that expects  this client, we would also need to specifically have it accept a `PrestoClient`:
```
func InitializePrestoExecutor(
	ctx context.Context,
	iCtx core.SetupContext,
	cfg *config.Config,
	resourceConfig map[string]int,
	prestoClient client.PrestoClient) (core.Plugin, error) {
```
Normally, I think this would be ok. But what I was trying to say (perhaps in a not so clear manner), is that if one day we wanted to abstract the client to be a more generic `ServiceClient` to use with not only Presto but also Hive, etc, then it's better if we rely on this interface (`ServiceClient`) as it would give us the flexibility on swapping out the client as we see fit. However, as I'm writing this, I realize that making this abstract is probably a larger refactoring effort that will likely change even this disregard this explanation haha. Sorry I think I'm rambling at this point